src/core_modules/capture-core/components/WidgetErrorAndWarning/content/WidgetErrorAndWarningContent.types.js

```diff
@@ -0,0 +1,20 @@
+// @flow
+import type { Rule, RuleObject } from '../WidgetError/WidgetError.types';
```

Comment:
1. I would define these types here instead of importing from WidgetError (and import the types defined here in WidgetError, so, the other way around)
2. I would change the naming here. The data that are passed in to the Widgets doesn't have to come from the rules engine and is really just a collection of messages (errors or warnings).